xarray/tests/test_groupby.py

```diff
@@ -30,7 +30,7 @@ def dataset():
             "foo": (("x", "y", "z"), np.random.randn(3, 4, 2)),
             "baz": ("x", ["e", "f", "g"]),
         },
-        {"x": ["a", "b", "c"], "y": [1, 2, 3, 4], "z": [1, 2]},
+        {"x": ("x", ["a", "b", "c"], {"name": "x"}), "y": [1, 2, 3, 4], "z": [1, 2]},
```

Comment:
I assume there is a `assert_identical` somewhere below?

Reply:
Yes some existing tests failed when I added this.